### ISSUE TITLE
Cannot change from C to F in UI

### ISSUE THREAD

**Issue by psmith3** [georgezhao2010/midea_ac_lan#18]
I saw similar closed issues about not being able to get temp to show up in F and I have tried changing the actual device from F to C and back to F, made a temp change on the unit, but Lovelace and the entity for the AC and outside temp shows us in C. Is this a bug or is there a work around? Just installed latest version this morning.

**Comment by georgezhao2010** [comment]
> I saw similar closed issues about not being able to get temp to show up in F and I have tried changing the actual device from F to C and back to F, made a temp change on the unit, but Lovelace and the entity for the AC and outside temp shows us in C. Is this a bug or is there a work around? Just installed latest version this morning.

I can't tell I'm quite understand what you mean.

Let me explain the process of temperature uint settings in this integration.

When the integration was set up (eg. HA starting up), it would follow HA's uints setting (in Settings->System->General) and try to set your AC's temperature unit and take AC's feedback as the final settings.

So when you change the setting of C or F in HA any time but starting up, the integration won’t set it to AC again.

Also some AC maybe not support C or F, no matter what you set AC always returns to which one it supports.

You could restart HA after setting uint (to C or F) to see if the units of UI and device is what you wanted.

**Comment by kriskirkland** [comment]
I'm having a similar issue; Upon first setup the Thermostat card was displaying in "F", then after a while it started displaying in "C" every couple of updates. After about an hour it gave up and now only shows "C" units. The AC itself is set to "F" and displays correctly on the front panel. I have tried changing the sensor options via the ui to "F" but as soon as you click save/update it goes back. I suspect this may be an issue with the Thermostat card, but I have my "imperial" units set globally.

**Comment by georgezhao2010** [comment]
> I'm having a similar issue; Upon first setup the Thermostat card was displaying in "F", then after a while it started displaying in "C" every couple of updates. After about an hour it gave up and now only shows "C" units. The AC itself is set to "F" and displays correctly on the front panel. I have tried changing the sensor options via the ui to "F" but as soon as you click save/update it goes back. I suspect this may be an issue with the Thermostat card, but I have my "imperial" units set globally.

Could you turn debug on and capture logs when the displayed C or F changes?

**Comment by Wferr** [comment]
Having the same problem, Here is when it swaps when changing temp. Thanks

```
2022-07-12` 23:50:11 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Device [xxxxxxxxxxxxx] received original message as hex 8370008e20636bf79a401b97c57089182b56156b85f9377107c89e5132231fdf30aac61751567eb7b37b12f572bc738039f5b0908213a0c940484fdd50497c69860372547f4dc01ae62528ab30dde29fc01ef27c460ac9467edf8d5dabbbfa91fb5dd3d213673ad37c965bbf830ae344fce5bd5e55f4ffb59f5b203c7d52de19caf2cfc6f7a7e15c1968f8f9f629d883df735ca3d0c7
2022-07-12 23:50:11 DEBUG (Thread-3) [custom_components.midea_ac_lan.midea.message_parser] Parsing command: aa23ac00000000000302c00139660000003c0010045f5600000000000000000000087748
2022-07-12 23:50:11 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Received message from device [xxxxxxxxxxxxx]: {'header': 'aa23ac00000000000302', 'body': 'c00139660000003c0010045f560000000000000000000008', 'type': '0x2c0'}
2022-07-12 23:50:11 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Device [xxxxxxxxxxxxx] status update to: {'available': True, 'power': True, 'mode': 1, 'fan_speed': 102, 'swing_vertical': True, 'swing_horizontal': False, 'target_temperature': 25.5, 'indoor_temperature': 22.0, 'outdoor_temperature': 18.0, 'comfort_mode': False, 'eco_mode': True, 'aux_heat': False, 'temp_fahrenheit': True, 'indirect_wind': False}
2022-07-12 23:50:12 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Device [xxxxxxxxxxxxx] received original message as hex 8370008e2063627000cd1befde527f7f28c62aa031c8c42d60f3f379a0304bfca696a2c36357c44114cda28fc4a76e75d6c5adccf05086833bf460ade760ef990f9decab52adc60312532cc4ef31d6401c5e70d39775610f6fb54705a4889704841d63e4d2606b71108d96399884acca95fff9f81ed65a6693242a89b168fa3d1a96a5dc0f6e887c8965db48d3301933a8467a27fcea
2022-07-12 23:50:12 DEBUG (Thread-3) [custom_components.midea_ac_lan.midea.message_parser] Parsing command: aa22ac00000000000305a05b20660000003c009000000000000000000000000013a723
2022-07-12 23:50:12 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Received message from device [xxxxxxxxxxxxx]: {'header': 'aa22ac00000000000305', 'body': 'a05b20660000003c009000000000000000000000000013', 'type': '0x5a0'}
2022-07-12 23:50:12 DEBUG (Thread-3) [custom_components.midea_ac_lan.state_manager] Device [xxxxxxxxxxxxx] status update to: {'available': True, 'power': True, 'mode': 1, 'fan_speed': 102, 'swing_vertical': True, 'swing_horizontal': False, 'target_temperature': 25.5, 'comfort_mode': False, 'eco_mode': True, 'aux_heat': False, 'temp_fahrenheit': False, 'indirect_wind': False}
```

**Comment by georgezhao2010** [comment]
@psmith3 @kriskirkland @Wferr 
Fixed in [v0.1.18](https://github.com/georgezhao2010/midea_ac_lan/releases/tag/v0.1.18)

Update and see if it works.
